src/Photons.ts

```diff
@@ -86,22 +77,26 @@ export default class Photons {
    * @param i Photon id, from [0, ..., nPhotons - 1]
    * @returns E.g. for 1: [4, 5]
    */
-  vectorPosIndicesForParticle(i: number): number[] {
+  vectorPosIndicesForPhoton(i: number): number[] {
     return [4 * i, 4 * i + 1]
   }
 
   /**
    * Create a single photon vector.
+   * @param IPolarization inteface containing:
    * @param posX Position of the photon, x.
    * @param posY Position of the photon, y.
-   * @param dirDirection Direction from ['>', '^', '<', 'v].
-   * @param pol Polarization from ['H', 'V'].
+   * @param direction Direction in degrees converted to ['>', '^', '<', 'v].
+   * @param polarization Polarization in complex values converted to ['H', 'V'].
    *
    * @returns A vector [dimX, DimY, dir, pol], does not modify the object.
+   * @todo Create converter helper functions for direction and polarization allowing more values.
    */
-  createPhoton(posX: number, posY: number, dir: string, pol: string): Vector {
+  createPhoton(photon: IPolarization): Vector {
     const dimensions = [this.dimX, this.dimY, Dimension.direction(), Dimension.polarization()]
-    const state = [posX.toString(), posY.toString(), dir, pol]
+    const directionStr = angleToStr(photon.direction)
+    const polarizationStr = convertPolarizationToStr(photon.polarization)
+    const state = [photon.x.toString(), photon.y.toString(), directionStr, polarizationStr]
```

Comment:
A big change, not backward compatible change (we use it a lot in different places).

Create a different function instead.

Also: I am personally against `direction Direction in degrees converted to ['>', '^', '<', 'v].` Such things such be discussed beforehand.

Reply:
I think it's nice to be able to create a photon with any polarization using an already available interface. It might be better in a new function.